eth/client.go

```diff
@@ -887,6 +918,32 @@ func (c *client) ReplaceTransaction(tx *types.Transaction, method string, gasPri
 	return newSignedTx, err
 }
 
+func (c *client) WatchForJob(streamId string) (*contracts.JobsManagerNewJob, error) {
+	ctx, _ := context.WithTimeout(context.Background(), c.txTimeout)
+	for {
+		sink := make(chan *contracts.JobsManagerNewJob)
+		sub, err := c.JobsManagerSession.Contract.JobsManagerFilterer.WatchNewJob(nil, sink, []ethcommon.Address{c.Account().Address})
```

Comment:
I think this might run into the same issue we did for Infura in `eventmonitor.go`, and ended up having to add the retry logic (anywhere we used `backoff.Retry`)

Reply:
Sure. Note that this already runs in a for-loop and retries on subscription errors, but not context timeouts. I can change it to cover both cases within`backoff.Retry` to be more consistent with the rest of the code.

One thing I dislike about `backoff.Retry` is that the API is clunky due to golang. In order to extract a non-error value from `Retry`, it forces us into the pattern of: declare an externally scoped variable, runtime create a closure for the var, explicitly mutate the var, the finally return that after error checking. With polymorphic HOFs, we could replace the `for { }` with a `func` and be done with it.

Reply:
Okay, the changes are in. Note that the watcher API has changed since the review -- now we do the job lookup inside this watcher and return the `Job` directly. I've opted not to retry the `GetJob` call if that fails, and instead we reconstruct the job from the event data, with the caveat that some fields may be missing. Whether any of that is a good idea, is open to discussion.

https://github.com/livepeer/go-livepeer/blob/948fe4b2971ddacd485f0ace392f101fc9091af8/eth/client.go#L953-L973